src/sentry/profiles/flamegraph.py

```diff
@@ -176,3 +189,172 @@ def extract_profile_ids() -> list[str]:
+    )
+    data = raw_snql_query(
+        request,
+        referrer=Referrer.API_PROFILING_PROFILE_FLAMEGRAPH.value,
+    )["data"]
+    spans: dict[str, list[IntervalMetadata]] = {}
+    for row in data:
+        if row["profiler_id"] in spans:
+            spans[row["profiler_id"]].append(
+                {
+                    "active_thread_id": row["active_thread_id"],
+                    "start": row["start_timestamp_precise"],
+                    "end": row["end_timestamp_precise"],
+                }
+            )
+        else:
+            spans[row["profiler_id"]] = [
+                {
+                    "active_thread_id": row["active_thread_id"],
+                    "start": row["start_timestamp_precise"],
+                    "end": row["end_timestamp_precise"],
+                }
+            ]
+    return spans
+
+
+class SpanMetadata(TypedDict):
+    profiler_id: list[IntervalMetadata]
+
+
+def get_chunk_snuba_conditions_from_spans_metadata(
+    spans: dict[str, list[IntervalMetadata]],
+) -> list[BooleanCondition | Condition]:
+    cond = []
+    for profiler_id, intervals in spans.items():
+        chunk_range_cond = []
+        for interval in intervals:
+            start = interval.get("start")
+            end = interval.get("end")
+            chunk_range_cond.append(
+                And(
+                    [
+                        Condition(Column("end_timestamp"), Op.GTE, start),
+                        Condition(Column("start_timestamp"), Op.LT, end),
+                    ],
+                )
+            )
+        cond.append(
+            And(
+                [
+                    Condition(Column("profiler_id"), Op.EQ, profiler_id),
+                    Or(chunk_range_cond) if len(chunk_range_cond) >= 2 else chunk_range_cond[0],
+                ]
+            )
+        )
+    return [Or(cond)] if len(cond) >= 2 else cond
+
+
+def get_chunks_from_spans_metadata(
+    organization_id: int,
+    project_id: int,
+    spans: dict[str, list[IntervalMetadata]],
+) -> list[dict[str, Any]]:
+    query = Query(
+        match=Storage(StorageKey.ProfileChunks.value),
+        select=[
+            Column("profiler_id"),
+            Column("chunk_id"),
+        ],
+        where=[Condition(Column("project_id"), Op.EQ, project_id)]
+        + get_chunk_snuba_conditions_from_spans_metadata(spans),
+        limit=Limit(100),
+    )
+    request = Request(
+        dataset=Dataset.Profiles.value,
+        app_id="default",
+        query=query,
+        tenant_ids={
+            "referrer": Referrer.API_PROFILING_PROFILE_FLAMEGRAPH.value,
+            "organization_id": organization_id,
+        },
+    )
+    data = raw_snql_query(
+        request,
+        referrer=Referrer.API_PROFILING_PROFILE_FLAMEGRAPH.value,
+    )["data"]
+    chunks = []
+    for row in data:
+        intervals = [
+            {
+                "start": str(int(datetime.fromisoformat(el["start"]).timestamp() * 10**9)),
```

Comment:
Are we still expecting timestamps in nanoseconds in vroom? I thought the continuous profiling chunks stored timestamps in seconds so there shouldn't be a need to do additional conversion here.

Reply:
This is for the callTrees we generate with the chunks. Basically instead of copying and pasting a ton of code that would be exactly the same and that the only diff would be the timestamp, I'm reusing all the existing Calltree/Node/Flamegraph code. The price to pay for reusing all this stuff is this conversion though.